Updated recipe for version sentence-transformer version 2.1.0

sugatoray opened the pull request:
The following are the list of activities covered in this PR. The absence of `torchvision` library in the feedstock recipe was causing problem with [another PR](https://github.com/conda-forge/staged-recipes/pull/16963). Please merge this PR (once it passes all the CI's).

This closes Issue #22. 

- [x] updated sha256
- [x] added dependencies:
	- [x] `torchvision`
	- [x] `tokenizers`
	- [x] `sentencepiece`
	- [x] `huggingface_hub`
- [x] updated/corrected versions of:
	- [x] `pytorch`
	- [x] `transformers`
- [x] updated test (pip check)
- [x] added @sugatoray as a maintainer

<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged) `NA` (changed version: `2.1.0`)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

sugatoray committed:
updated recipe: version 2.1.0
- updated sha256
- added dependencies:
- torchvision
- tokenizers
- sentencepiece
- huggingface_hub (as run_constrained)
- updated/corrented versions of:
  - pytorch
  - transformers
- updated test (pip check)
- added sugatoray as a maintainer

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

sugatoray committed:
corrected python version spec format

sugatoray committed:
fixed version spec format for all deps

sugatoray commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 3.21.6, conda-smithy 3.14.3, and conda-forge-pinning 2021.11.18.16.59.41

sugatoray commented:
cc: @ndmaxar, @oblute, @rluria14

sugatoray commented:
gentle reminder: @ndmaxar, @oblute, @rluria14

BastianZim commented on recipe/meta.yaml:
```diff
@@ -19,14 +19,22 @@ requirements:
+    - sentencepiece
+    - tokenizers >=0.10.3
+    - transformers >=4.6.0,<5.0.0
+  run_constrained:
+    # NOTE: huggingface-hub is not yet available on conda-forge.
+    # Moving huggingface-hub under "run_constrained" as a temporary 
+    # fix, until the package is made available on conda-forge.
+    - huggingface-hub
```
```suggestion
huggingface_hub
```

The package is on conda with `_` instead of `-`.

sugatoray replied:
Good catch! Thanks.

sugatoray committed:
Update recipe/meta.yaml

huggingface-hub is available on conda-forge as huggingface_hub
- https://anaconda.org/conda-forge/huggingface_hub

Co-authored-by: Bastian Zimmermann <10774221+BastianZim@users.noreply.github.com>

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * Failed to even lint the recipe, probably because of a conda-smithy bug :cry:. This likely indicates a problem in your `meta.yaml`, though. To get a traceback to help figure out what's going on, install conda-smithy and run `conda smithy recipe-lint .` from the recipe directory.

BastianZim commented on recipe/meta.yaml:
```diff
@@ -19,14 +19,18 @@ requirements:
+    - sentencepiece
+    - tokenizers >=0.10.3
+    - transformers >=4.6.0,<5.0.0
+huggingface_hub
```
```suggestion
    - huggingface_hub
```
That was my bad, sorry.

sugatoray replied:
No problem. I took care of it. :)

sugatoray committed:
fixed indentation

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

sugatoray commented:
@conda-forge-admin, please ping conda-forge/core

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I was asked to ping @conda-forge/core and so here I am doing that.
